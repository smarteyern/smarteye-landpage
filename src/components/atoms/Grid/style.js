import styled, { css } from "styled-components";
import { breakAt, BreadkpointsSize } from "styles/Breadkpoints";

const getPrimaryLight = ({ theme }) => theme.colors.primary.light;

const breakColumnAt = (size) => (props) =>
  props[size] &&
  css`
    ${breakAt(BreadkpointsSize[size])} {
      grid-template-columns: repeat(${(props) => props[size]}, 1fr);
    }
  `;

export const GridRoot = styled.div`
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;

  ${breakColumnAt("sm")}

  ${breakColumnAt("md")}
  
  ${breakColumnAt("lg")}

  ${breakColumnAt("xl")}
`;

export const Box = styled.div`
  height: 200px;
  background-color: ${getPrimaryLight};
`;
