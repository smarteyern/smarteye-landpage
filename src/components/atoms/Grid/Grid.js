import React from "react";
import PropTypes from "prop-types";
import { GridRoot, Box } from "./style";

const Grid = () => (
  <GridRoot sm={2} md={3} lg={4} xl={5}>
    <Box></Box>
    <Box></Box>
    <Box></Box>
    <Box></Box>
    <Box></Box>
  </GridRoot>
);

Grid.defaultProps = {
  xs: undefined,
  sm: undefined,
  md: undefined,
  lg: undefined,
  xl: undefined,
  xxl: undefined,
};

Grid.propTypes = {
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
  xl: PropTypes.number,
  xxl: PropTypes.number,
};

export default Grid;
