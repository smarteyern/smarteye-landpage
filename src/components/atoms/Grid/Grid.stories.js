import React from "react";
import Grid from "./Grid";

export default {
  title: "Components/Atoms/Grid",
  component: Grid,
};

export const usage = () => <Grid></Grid>;
