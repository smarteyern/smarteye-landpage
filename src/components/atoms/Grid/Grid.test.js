import React from "react";
import { render } from "test-utils";

import Grid from "./Grid";

test("Match snapshot with no params", () => {
  const { asFragment } = render(<Grid></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("Match snapshot with params", () => {
  const { asFragment } = render(<Grid sm={2} md={3} lg={4} xl={5}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("Match snapshot with params sm", () => {
  const { asFragment } = render(<Grid sm={1}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("Match snapshot with params md", () => {
  const { asFragment } = render(<Grid md={4}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("Match snapshot with params lg", () => {
  const { asFragment } = render(<Grid lg={8}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});

test("Match snapshot with params xl", () => {
  const { asFragment } = render(<Grid xl={10}></Grid>);

  expect(asFragment()).toMatchSnapshot();
});
