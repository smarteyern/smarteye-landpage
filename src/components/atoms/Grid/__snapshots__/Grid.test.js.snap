// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Match snapshot with no params 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports['Match snapshot with params 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports['Match snapshot with params lg 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports['Match snapshot with params md 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports['Match snapshot with params sm 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;

exports['Match snapshot with params xl 1'] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-column-gap: 16px;
  grid-row-gap: 16px;
  grid-template-columns: 1fr;
}

.c1 {
  height: 200px;
  background-color: #56CCF2;
}

@media (min-width:576px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: repeat(4,1fr);
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-template-columns: repeat(5,1fr);
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
